### DISCUSSION TITLE
一些功能需求调查

### DISCUSSION THREAD

**Discussion by noberumotto** [Planshit/Tai#79] [Polls]
想看看大家需要哪些功能，如果没有你需要的可以回复

**Comment by Luciushy** [comment]
有没有想过一个问题，像spotify这样的应用在大多数时候都是后台运行的，更别说还有flux、ditto这些了，是否可以针对某些特定应用统计其运行时间而非焦点所在？

**Reply by noberumotto** [reply]
做这个软件一开始的初衷是仅仅考虑统计前台有焦点的软件使用时长，后台当时是没考虑的。现在会考虑加入开发计划中

**Comment by NoclubOIRzXx** [comment]
给软件重命名和给软件上图标。就像文件夹里面的自定义图标，可否实现？

**Reply by noberumotto** [reply]
可以的。

**Comment by AsDaShu** [comment]
希望可以统计每个网站的使用时间，之前我用的一款软件manicTime软件是可以统计网站的使用时间的 ，这样自己复盘的时候可以知道自己在什么网站上用了多少时间，是学习还是摸鱼浪费时间了

**Reply by noberumotto** [reply]
已经在开发中了

**Reply by AsDaShu** [reply]
这个功能大概什么时候能上线

**Reply by noberumotto** [reply]
具体不好说，目前的进度是能够简单统计chrome的记录了，还需要完善和测试以及增加一个edge浏览器的支持后才会发布，可能得春节之后了

**Comment by 2710128764** [comment]
大佬可以做一个选项让鼠标在不操作多长时间后就不统计使用时间的功能吗，谢谢啦。

**Reply by noberumotto** [reply]
这个功能一直都有，不过不可调整，默认是5分钟左右没有鼠标操作且没有声音在播放就会停止统计。

**Reply by 2710128764** [reply]
略略略，明白了，老哥新年快乐。

**Comment by cstarpc** [comment]
大佬新年快乐，有一个问题，我发现在统计列表里，按天/月/年都自己选择回溯时间，但是在**按周**的列表里，只能选本周和上周，可不可以加一个自己选择周数回溯的功能呀？
另外，在详细列表里，没有按周的选项，可不可以加呢0.0

**Comment by Edward-L** [comment]
可以在分类里搞一个未分类的组吗？把没有分类的都归纳进去，这样比较方便去看有哪些没有分类的，方便后续分类，现在都要去统计里看哪个是没被分类但是有时长的。强迫症看到存在未分类浑身难受。。。

**Reply by majianchuan** [reply]
欢迎试试我开发的：https://github.com/majianchuan/ScreenTimeTracker

**Comment by hhht110** [comment]
这个软件 感觉很不错 希望未来不要加入广告啥的 哈哈哈。挺有意思的一个软件。可以考虑捐赠比如我可以捐赠一瓶红牛供作者熬夜写代码的时候食用 哈哈

**Reply by noberumotto** [reply]
谢谢，永远不会有广告的

**Comment by ioqbit** [comment]
![image](https://user-images.githubusercontent.com/66349171/211151754-aaf5146c-ed15-49eb-969f-336c139d4a5a.png)
能否增加一个按照具体时间显示记录的界面。比如上午用PPT做的是一个项目，下午用PPT做另一个项目，我想知道分别用了多少时间，而目前的功能只能看到每小时内分类的时间

**Comment by Jim123666888** [comment]
现在的Tia只能记录我电脑的屏幕时间，我觉得可以增加一些开始和停止按钮来记录屏幕外的时间，比如吃饭，睡觉，读书，学习，刷手机等等。按下按钮或者手动登记时间就可以把行为记录并整合进Tia里，并把它和应用使用的时间放在一起做成时间轴，这样我就可以更直观地看见自己的时间花在了哪里。
还有现在的时间轴不能按时间先后来显示我的行为，我觉得可以改成按时间先后来显示色块。同时只要分类一多，色块在表格里总是挤成一块看不清，我觉得加个可以增加个时间条的高度的设置来把时间条拉长，或者把时间条横过来利用屏幕横向空间就可以解决这一问题。

**Reply by noberumotto** [reply]
这个需求需要仔细考虑，以及更多人的意见。有些超出Tai的使命了

**Reply by ocean-sudo** [reply]
这种纯记录作息的软件手机端和手表端倒是挺多的，我觉得在电脑上目前没有多端的想法来看不太有特别的必要性。
毕竟，设想一下场景，你吃完饭，玩了会儿手机，等过了会再去电脑前记录两者时间，这种是不太现实的。
我建议你可以找找类似的手机软件，做整理时两者相结合进行判断。

**Reply by zyjdmmm** [reply]
你只要知道你每天工作总长度就行了吧，知道那么细感觉没啥用啊，你会去考虑上个月15号早上吃饭用了多久么，你只需要知道那天工作了多久就大概明白了，你如果非需要手动的，推荐番茄todo，或toggle track（虽然我都不太喜欢）

**Comment by 0xWelt** [comment]
能否考虑引入抽象的“行为”概念，通过进程的联动规则（某些进程同时存在且某些进程不存在时）来触发一个标记行为？这样的好处是让用户有更强的控制记录粒度的能力。举一个例子，当我同时打开“zotero”与“notion”，且其中有一个窗口活跃时，则正在进行“看论文”这一行为，同时可以选择性的存在“xx音乐”但并不影响这段时间的行为属性。
现在的“分类”统计模式可以看作一种一元规则，即：某进程前台活跃 -> 正在执行该进程分类对应的行为。
可以通过可视化的方式展现同时存在的进程（在一个时间轴上），方便用户快速标注某段时间的行为，并自动复用规则。

**Comment by cangni** [comment]
> 想看看大家需要哪些功能，如果没有你需要的可以回复

不同电脑记录的数据支持合并吗，现在只有导出功能

**Comment by goutenglee** [comment]
有两个UI上的小建议。

1.“应用/网页”改成一键切换。
现在切换模式，要先点一下出现下拉菜单，再点一下才能切换到另一个，一共要点两次鼠标，并且还要移动鼠标去选。
下拉菜单适用于四五个以上的选项的切换，现在才两个选项的切换，没必要用下拉菜单。
![image](https://github.com/user-attachments/assets/fc0fed5e-ead8-4584-9c1b-0681fdda1784)

鼠标点一下就切到另一个，比较方便。
比如把这个下拉菜单按钮，里面那个三角形改成循环圆圈，暗示用户点一下能切换，达成一键切换的效果。
![image](https://github.com/user-attachments/assets/85357627-b644-4964-bd17-1a8a865f2c38)

或者使用文字按钮，改成“应用/网页”，将未选中的那个显示成灰色的就可以，类似这样（我没改灰度）：
![image](https://github.com/user-attachments/assets/4844c440-2532-44d5-a035-0c9578c232f5)


2.增加“上一天 下一天”、“上一周 下一周”这样的按钮。
现在要查看昨天的情况，需要点进日历去选，略微麻烦。
多数人只会切换一下看最近几天的情况，最好增加一下。不过日历选择具体日期别去掉，一定要保留。
就像这样加两个按钮：
![image](https://github.com/user-attachments/assets/25a26ef8-f90b-416b-94ba-1f4c07d7ba58)

**Comment by zyjdmmm** [comment]
本来都准备自己开发了，幸好最后找到了Tai，我要说的是尽量功能简洁些
，以下这些软件我研究了一天，很多软件功能都过于复杂压根连按钮都不知道在哪里😅😅😅，可惜我主要是开发qt的c#还不太会。还有我要吐槽手机上那个番茄TODO，也是加了一吨功能导致我现在都不想用。我就是想要简简单单看看每天在各个事情上花了多少时间
![image](https://github.com/user-attachments/assets/b76f1695-9174-43c4-ac91-db6868877b27)
![image](https://github.com/user-attachments/assets/face4203-b666-4059-9e0e-c277007d6121)

**Reply by majianchuan** [reply]
欢迎试试我开发的：https://github.com/majianchuan/ScreenTimeTracker

**Comment by lswx822** [comment]
你好，想问下，怎么把统计时间里软件自己的使用时间给去掉啊？我添加Tai，或者Tai.exe，统计表里面还是有对tai的使用时间的记录

**Reply by zyjdmmm** [reply]
这个是合理的吧，有对tai的使用时间的记录就是说明当天你对tai在那捣消耗了时间

**Reply by lswx822** [reply]
不是啊，tai本来就有个白名单，但是按软件要求也没办法屏蔽统计我才问下，说不定是我自己操作的问题

**Comment by Canlan263** [comment]
时间筛选的时候能不能加个近7天，或者自由选择筛选区间，先选开始时间再选结束时间这样统计。

**Comment by lswx822** [comment]
你好，已收到，辛苦

**Comment by Stargazed-Dreamer** [comment]
想要稍微精细一点的记录显示，即时间线功能。现在是按小时来显示的，想要一个界面：横轴为时间，纵轴为【焦点应用/焦点应用类别】这样的功能，方便对今日回顾🙏🙏

**Comment by lswx822** [comment]
你好，已收到，辛苦

**Comment by lswx822** [comment]
你好，已收到，辛苦

**Comment by YQ157** [comment]
希望增加当天开始时间的设置。很多人都不在0点前睡觉，希望出一个设置项，可以设置“开启新一天统计”的时间。因为体感上，即使过了0点，但是没睡觉，就还是“当天”，而下次醒来才是“新的一天”。
谢谢！

**Comment by lswx822** [comment]
你好，已收到，辛苦
